spine_json_lib/deserializer/spine_nodes.py

```diff
@@ -157,112 +164,134 @@ def to_json_data(cls, graph: DAGraph) -> Dict[str, Any]:
             ]
         return json_result
 
+    @staticmethod
+    def add_bone_to_graph(graph, bone_data, idx):
+        node_data = SpineNodeData(
+            node_data=bone_data,
+            node_type=NodeType.BONE,
+            node_base_id=bone_data["name"],
+            idx=idx,
+        )
+        graph.add_node(
+            node_type=node_data.node_type,
+            node_id=node_data.node_id,
+            node_data=node_data.node_data,
+        )
+        # Getting optional parent from bone
+        parent_id = SpineGraphParser._to_graph_id(
+            node_type_name=NodeType.BONE.name, node_base_id=bone_data.get("parent")
+        )
+        parent = graph.get_node(parent_id)
+        if parent:
+            graph.add_edge(parent.id, node_data.node_id)
+        return node_data
+
+    @staticmethod
+    def add_slot_to_graph(graph: DAGraph, slot_data: Dict[str, Any], idx: int):
+        spine_slot_data = SpineNodeData(
+            node_data=slot_data,
+            node_type=NodeType.SLOT,
+            node_base_id=slot_data["name"],
+            idx=idx,
+        )
+
+        graph.add_node(
+            node_type=spine_slot_data.node_type,
+            node_id=spine_slot_data.node_id,
+            node_data=spine_slot_data.node_data,
+        )
+
+        # Getting optional parents from slot
+        parent_id = SpineGraphParser._to_graph_id(
+            node_type_name=NodeType.BONE.name, node_base_id=slot_data.get("bone")
```

Comment:
is node_type_name BONE or SLOT?

Reply:
In here we are adding SLOT node as child to a BONE.
So in this line we are getting the "bone" information from the slot and generating the internal "node_id".